libcontainer/container_linux.go

```diff
@@ -522,6 +522,27 @@ func (c *linuxContainer) commandTemplate(p *Process, childInitPipe *os.File, chi
 	return cmd
 }
 
+// shouldSendMountSources says whether the child process must setup bind mounts with
+// the source pre-opened (O_PATH) in the host user namespace.
+// See https://github.com/opencontainers/runc/issues/2484
+func (c *linuxContainer) shouldSendMountSources() bool {
+	// Passing the mount sources via SCM_RIGHTS is only necessary when
+	// both userns and mntns are active.
+	if len(c.config.Mounts) == 0 {
+		return false
+	}
+	if !c.config.Namespaces.Contains(configs.NEWUSER) ||
+		!c.config.Namespaces.Contains(configs.NEWNS) {
+		return false
+	}
+	// nsexec.c send_mountsources() requires setns(mntns) capabilities
+	// CAP_SYS_CHROOT and CAP_SYS_ADMIN
+	if c.config.RootlessEUID {
```

Comment:
I think it makes sense to move this check to before `if !c.config.Namespaces.Contains(...` as it is faster.

Reply:
I moved it because the linter was crying if the last part of the code is just:

```
        if !bindMounts {
                return false
        }

        return true
```

As it thinks I should just `return !bindMounts` or something like that, and I think that is not better. I like the current structure of "check if something to make it false is there, return true if we checked all already". 

TBH, I doubt of the perf benefit of changing the checks order. But sure, moving the check up again and let's see if I find a way to make the linter and myself happy :)

Reply:
Well, the linter was not wrong, `return bindMounts` is quite fine too (not sure why suggested !!bindMounts, but returning just bindMounts makes it happy too :))